### ISSUE TITLE
SQLiteDiskIOException opening collection after sync

### ISSUE THREAD

**Issue by cameel** [ankidroid/Anki-Android#3607]
I was doing reviews on my tablet, finished a few decks, decided to sync and left the tablet for a while. Then I came back and I saw the error reporting dialog. So it must have crashed because of an exception.

It was the third of fourth sync today. I let Android update a few apps in the background and the update might have been still running while I was syncing. I don't know if it has any relation to the crash.

Sorry for an overly generic title of this bug report but AnkiDroid did not show any details of the exception and I don't know what to say about it apart from the fact that it happened. I can't even search for similar issues. I have sent the report from the app so you can look at it and see what the error was.

```
AnkiDroid Version = 2.4.3
Android Version = 5.1.1
ACRA UUID = 815a9261-620e-4280-a814-f075aae6be54
```

**Comment by timrae** [comment]
Please paste your debug info from the about dialog

**Comment by cameel** [comment]
I did.

**Comment by timrae** [comment]
Ah thanks, it wasn't showing up in the email notification but I can see it on the web version

**Comment by cameel** [comment]
Yeah, sorry for that. I have edited the post immediately when I noticed that it's missing but github apparently does not send any notifications about this.

**Comment by timrae** [comment]
android.database.sqlite.SQLiteDiskIOException: disk I/O error (code 1034)
at android.database.sqlite.SQLiteConnection.nativeExecuteForCursorWindow(Native Method)
at android.database.sqlite.SQLiteConnection.executeForCursorWindow(SQLiteConnection.java:845)
at android.database.sqlite.SQLiteSession.executeForCursorWindow(SQLiteSession.java:836)
at android.database.sqlite.SQLiteQuery.fillWindow(SQLiteQuery.java:62)
at android.database.sqlite.SQLiteCursor.fillWindow(SQLiteCursor.java:144)
at android.database.sqlite.SQLiteCursor.getCount(SQLiteCursor.java:133)
at android.database.AbstractCursor.moveToPosition(AbstractCursor.java:197)
at android.database.AbstractCursor.moveToNext(AbstractCursor.java:245)
at com.ichi2.anki.AnkiDb.queryString(AnkiDb.java:167)
at com.ichi2.anki.AnkiDb.setWalJournalMode(AnkiDb.java:385)
at com.ichi2.anki.AnkiDb.<init>(AnkiDb.java:70)
at com.ichi2.anki.AnkiDatabaseManager.getDatabase(AnkiDatabaseManager.java:44)
at com.ichi2.libanki.Storage.Collection(Storage.java:50)
at com.ichi2.anki.AnkiDroidApp.openCollection(AnkiDroidApp.java:539)
at com.ichi2.anki.AnkiDroidApp.openCollection(AnkiDroidApp.java:525)
at com.ichi2.async.CollectionLoader.loadInBackground(CollectionLoader.java:35)
at com.ichi2.async.CollectionLoader.loadInBackground(CollectionLoader.java:16)

**Comment by timrae** [comment]
(1034) SQLITE_IOERR_FSYNC

The SQLITE_IOERR_FSYNC error code is an extended error code for SQLITE_IOERR indicating an I/O error in the VFS layer while trying to flush previously written content out of OS and/or disk-control buffers and into persistent storage. In other words, this code indicates a problem with the fsync() system call in unix or the FlushFileBuffers() system call in windows.

**Comment by timrae** [comment]
It could be a hardware problem. Let us know if it happens again

**Comment by cameel** [comment]
OK. So far it happened just once.

**Comment by hssm** [comment]
Can we close this? Does it still happen?

**Comment by timrae** [comment]
I reckon close it as we no longer have that database manager class or set journal mode in 2.6 alpha. Please open a new issue if you experience it again

**Comment by cameel** [comment]
> Can we close this? Does it still happen?

I think it happened once more after that. But not in the last few months. I can't add anything substantial here to help trace it so I guess it's OK to close it.

**Comment by cameel** [comment]
It happened again just now. Or at least something similar, can't tell for sure because there were no error details. I have left the tablet to sync (after a few successful syncs earlier today) and there was a `Collection not opened` dialog that told me to check the database or restore from backup. I did the check and it said the database was OK. I synced and it succeeded.

It did not ask me to report the error (unless it does it automatically but it shouldn't since I have `Error reporting mode` set to `Ask me` in settings). Maybe because the application did not seem to actually crash (i.e. it wasn't closed and restarted).

It did happen this way (i.e. without a crash) before, maybe once or twice and I didn't report it because there's not much detail but now I think it might be related to this ticket. At least the circumstances when it happens are similar. I'm not sure if I should open a new issue for this because there's not much detail (not even an error report) and I'm still on AnkiDroid 2.5 so I'll just leave this here.

BTW, this still on the same Nexus 7 (2013).

```
AnkiDroid Version = 2.5.4
Android Version = 5.1.1
ACRA UUID = aeee41c2-6522-4cb1-aaba-4cf0a8c6c06e
```

**Comment by timrae** [comment]
Thanks. I'll keep the issue closed as you're still on 2.5.4 and we've made some big changes in 2.6 which are pretty likely to have fixed the issue.

From 2.5, we don't send repeat reports anymore, which is probably why you didn't see a "sending report" toast.
